doc/policies.md

```diff
@@ -0,0 +1,180 @@
+# APIcast policies
+
+The behaviour of APIcast is customizable via policies. A policy basically tells
+APIcast what it should do in each of the nginx phases. This document details
+how policies work and how you can write your own and integrate them into
+APIcast.
+
+- [**Policies**](#policies)
+- [**Policy chains**](#policy-chains)
+- [**APIcast default policies**](#apicast-default-policies)
+- [**Write your own policy**](#write-your-own-policy)
+- [**Integrate your policies**](#integrate-your-policies)
+
+
+## Policies
+
+A policy tells APIcast what it should do in each of the nginx phases: `init`,
+`init_worker`, `rewrite`, `access`, `balancer`, `header_filter`, `body_filter`,
+`post_action`, and `log`.
+
+Policies can share data between them. They do that through what we call the
+`context`. Policies can read from and modify that context in every phase.
+
+
+## Policy chains
+
+Policies can be combined using a policy chain. A policy chain is simply a
+sorted list of policies.
+
+The way policy chains work is as follows: suppose that we have a policy A that
+describes what to do in the `rewrite` and `header_filter` phases and a policy B
+that describes what to run in `access` and `header_filter`. Assume also that
+when describing the chain, we indicate that policy A should be run before
+policy B. When APIcast receives a request, it will check the policy chain
+described to see what it should run on each phase:
+- rewrite: execute what policy A specifies for this phase.
```

Comment:
"execute the method/function policy A provides for this phase" or similar?

Reply:
:white_check_mark: